gridt/src/app/core/settings.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from "@angular/core";
+import { Observable, pipe, throwError, of, iif } from "rxjs";
+import { flatMap, tap, map, catchError, pluck, mergeMap, take } from "rxjs/operators";
+
+import { Identity } from './models/identity.model';
+import { AuthService } from './auth.service';
+import { SecureStorageService } from './secure-storage.service';
+
+@Injectable({
+  providedIn: "root"
+})
+
+export class SettingsService {
+
+  constructor (
+    private auth: AuthService,
+    private secStore: SecureStorageService,
+  ) {}
+
+  /**
+   * Observable to obtain settings from secure storage
+   * If secure storage is empty, this returns an error
+   */
+  public localIdentity$: Observable<Identity> = this.auth.readyAuthentication$.pipe(
+    flatMap(() => this.secStore.get$("identity"))
+  );
+
+  /**
+   * Store settings in the secure storage.
+   * @param identity Settings to be stored.
+   */
+  public setLocalIdentity$(identity: Identity): Observable<boolean> {
+    return this.auth.isLoggedIn$.pipe(
+      take(1),
+      tap( (val) => {
+        if (val === true) {
+          console.debug(`Storing identity: ${JSON.stringify(identity)} into the local storage.`);
+          this.secStore.set$("identity", identity).subscribe();
+        }
+      }),
+      mergeMap( val => iif( () => val === true, of(true), throwError("Not logged in: can't store identity in local storage.")))
+    )
+  }
```

Comment:
This is what I would recommend to put in the pipe:
```ts
this.auth.isLoggedIn$.pipe(
  flatMap( (loggedIn) => {
    const setter = this.secStore.set$("identity", identity).pipe(
       catchError(() => throwError("Could not set identity: secure storage not available".))
    );
    return (loggedIn) ? setter: throwError("Could not set identity: not logged in");
  }), 
);
```
This has the following behaviour:
- If the user is not logged in, it throws an error.
- If the user is logged in, but the secure storage fails to set the identity, it throws an error.
- If the user is logged in and the secure storage sets the identity successfully, it will return the output of `secureStorage.set$`. (This "nexts" `true` in case the setting was successful.)

Reply:
If you'd go for this, I would recommend maintaining a error codes variable and drawing from there, but this was more for the idea.

Reply:
Great, I took the liberty to alter the code a little bit. I decided to use `readyAuthentication$` such that `secStore.set$` is never called when not logged in.